data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessor.java

```diff
@@ -79,34 +92,62 @@ public GeoIPProcessor(final PluginMetrics pluginMetrics,
   public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
     Map<String, Object> geoData;
 
+    final GeoIPDatabaseReader geoIPDatabaseReader = geoIPProcessorService.getGeoIPDatabaseReader();
+    final boolean databasesExpired = geoIPDatabaseReader.isExpired();
+
     for (final Record<Event> eventRecord : records) {
       final Event event = eventRecord.getData();
+
       final String whenCondition = geoIPProcessorConfig.getWhenCondition();
 
       if (whenCondition != null && !expressionEvaluator.evaluateConditional(whenCondition, event)) {
         continue;
       }
+      geoIpEventsProcessed.increment();
+
+      // TODO: Need to decide the behaviour, right now if all databases are expired we don't enrich the data.
+      if (databasesExpired) {
+        // TODO: Finalize the tags
+        continue;
+      }
 
       boolean isEventFailedLookup = false;
-      geoIpEventsProcessed.increment();
 
       for (final EntryConfig entry : geoIPProcessorConfig.getEntries()) {
         final String source = entry.getSource();
-        final List<String> attributes = entry.getFields();
-        final String ipAddress = event.get(source, String.class);
+        final List<GeoIPField> fields = entryFieldsMap.get(entry);
+        final Set<GeoIPDatabase> databases = entryDatabaseMap.get(entry);
+        String ipAddress = null;
+        try {
+          ipAddress = event.get(source, String.class);
```

Comment:
This is just a placeholder for future implementation?

Reply:
No, but as you mentioned above I will try to put all doExecute logic in try/catch in upcoming PR's